### DISCUSSION TITLE
"Headless" Pretix (extending APIs / documentation)

### DISCUSSION THREAD

**Discussion by fronbasal** [pretix/pretix#4291] [Ideas] [answered]
Dear Pretix Team,

are there any plans of allowing for third-party frontends to the Pretix ticketing system, similar to [Saleor](https://saleor.io/), [Shopify Headless Commerce](https://www.shopify.com/plus/solutions/headless-commerce) (see [vercel/commerce](https://github.com/vercel/commerce))?

How feasible would this be with the current API structure?

**Comment by raphaelm** [comment, marked as answer]
It is *possible* with a custom server-side component between your frontend and the API, but it requires you to rebuild *a lot* of pretix's logic in your application. It *has* been done (basically, that's how pretixPOS works, and we have some partner companies doing it), but it usually only is feasible to do if you restrict yourself on a tight use-case (e.g. "no add-ons, no discounts, only one payment provider") to reduce the scope of things to re-build.

Providing a "proper" headless shopping API is something we would be interested in as a strategic goal in the future, and we have talked about it internally before. However, I believe it would be an effort that takes many person-months or even -years to get right and it would require an entirely new API for this purpose. We have not prioritized this yet, so I wouldn't want to promise it for the forseeable future ;)

**Reply by fronbasal** [reply]
Thank you so much for your thorough answer, Raphael! Highly appreciated! 
Super interesting, that it was discussed internally - excited to see where Pretix will be headed mid-term!
